### PR TITLE
chore(sdk): drop Python 3.7 for kfp sdk

### PR DESCRIPTION
**Pull request by rickyxie0929**
**Description of your changes:**
chore(sdk): drop Python 3.7 for kfp sdk

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/RELEASE.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 ## Documentation updates
 * Make full version dropdown visible on all KFP SDK docs versions [\#10577](https://github.com/kubeflow/pipelines/pull/10577)
+* Drop support for Python 3.7 since it has reached end-of-life. [\#10734](https://github.com/kubeflow/pipelines/pull/10734)
```
Comment (connor-mccarthy) [inline sdk/RELEASE.md:14]: Number is stale

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]
I think the title of this PR is out-of-date since the changes are for the KFP SDK rather than GCPC SDK. Can we also use the [Pull Request Title Convention](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention)?

File: sdk/RELEASE.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 ## Documentation updates
 * Make full version dropdown visible on all KFP SDK docs versions [\#10577](https://github.com/kubeflow/pipelines/pull/10577)
+* Drop support for Python 3.7 since it has reached end-of-life(part1). [\#10750](https://github.com/kubeflow/pipelines/pull/10750)
```
Comment (chensun) [inline sdk/RELEASE.md:14]: I think this should be considered as a breaking change. Other than that, the change looks good to me.

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]
What is part 2 of this PR? I see the title says "(part 1)".

File: sdk/RELEASE.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## Features
 
 ## Breaking changes
+* Drop support for Python 3.7 since it has reached end-of-life(part1). [\#10750](https://github.com/kubeflow/pipelines/pull/10750)
```
Comment (connor-mccarthy) [inline sdk/RELEASE.md:6]: We can remove part1 from the change description

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**